Bluetooth: Controller: VEGA Board 2mbps support

alexandru-porosanu-nxp opened the pull request:
This PR adds support for the 2Mbps PHY on the VEGA Board

alexandru-porosanu-nxp committed:
bluetooth: controller: openisa/RV32M1: add 2 Mbps PHY support

The radio on the VEGA platform supports both 1Mbps, as well
as 2Mbps BLE PHYs. It does not support coded BLE PHY. This patch
adds the necessary callback, as well as timings to enable the 2 Mbps
PHY support in the SW LL HAL for VEGA.

Signed-off-by: Alex Porosanu <alexandru.porosanu@nxp.com>

alexandru-porosanu-nxp committed:
bluetooth: controller: enable 2Mbps PHY for RV32M1

This patch adds the selection of  the necessary CONFIG_*
options for allowing the use of the 2Mbps BLE PHY on
VEGA platform.

Signed-off-by: Alex Porosanu <alexandru.porosanu@nxp.com>

cvinayak reviewed (changes requested):
LGTM except the commit body with "supports the at both 1Mbps "

alexandru-porosanu-nxp commented:
> LGTM except the commit body with "supports the at both 1Mbps "

Right, sorry, I missed that ; I will resubmit ASAP

alexandru-porosanu-nxp commented:
And I forgot to update the index.rst accordingly. Will include that as well

alexandru-porosanu-nxp committed:
boards: riscv: rv32m1_vega: remove BLE 2Mbps limitation

Now the VEGABoard BLE support includes the support for
the 2 Mbps BLE PHY, so the docs need to be updated
accordingly.

Signed-off-by: Alex Porosanu <alexandru.porosanu@nxp.com>

alexandru-porosanu-nxp commented:
I have no idea why the doc check fails... @dbkinder, any ideas?

MaureenHelm commented:
> I have no idea why the doc check fails... @dbkinder, any ideas?

Please rebase

alexandru-porosanu-nxp commented:
> > I have no idea why the doc check fails... @dbkinder, any ideas?
> 
> Please rebase

Thanks Maureen, I just did, now let's see if the docs checking are passing now... From the logs I think it might've been a transient error in the infrastructure...

joerchan commented on subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c:
```diff
@@ -22,19 +22,26 @@
 #include "hal/ticker.h"
 #include "hal/swi.h"
 
+#define LOG_MODULE_NAME bt_openisa_radio
```
Commented about this line in other PRs as well. There should be a line that sets `BT_DBG_ENABLED` based on a kconfig before including common/log.h. otherwise the debug cannot be controlled through Kconfig.

#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER) for example.

alexandru-porosanu-nxp replied:
Right, I've resolved the conflict due to the encryption PR merged and will push soon

joerchan reviewed (commented):
Merge conflicts
